### PR TITLE
Login redirect on API auth failure

### LINKED ISSUE #23: Redirect user to login when unauthenticated
Currently, if unauthenticated, the user will still be able to go to all routes but the API calls will fail. It would be a better user experience to redirect the user to the login page automatically if they are not authenticated.

### PR DESCRIPTION
**Pull request by cabarnes**
Closes #23 

When a `401` is received from the API, redirect to the login page as the token has expired or is no longer valid.
This can be tested by going to mission control, clearing cookies, and trying to modify the program or the program name.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - lts/*
+  - "8"
```
[the remark this answers is not in the document]
Comment (cabarnes) [reply .travis.yml]: `10` is now the newest LTS. `8` will still be in LTS for another year.

**Comment by cabarnes** [comment]
Did you use social auth or username/password?

**Comment by aninternetof** [comment]
Username/password.

**Comment by cabarnes** [comment]
Ok. I was using social auth. I wonder if the redirects reset the state that doesn't happen with username/password

**Comment by cabarnes** [comment]
@aninternetof I think I've fixed the issue

**Comment by aninternetof** [comment]
Works great now!

**Pull request merged**